front/src/applications/operationalStudies/components/ManageTrainSchedule/PowerRestrictionSelector.tsx

```diff
@@ -10,23 +10,41 @@ import {
   getPowerRestriction,
   getPathfindingID,
 } from 'reducers/osrdconf/selectors';
-import { osrdEditoastApi, LightRollingStock } from 'common/api/osrdEditoastApi';
+import { osrdEditoastApi, RollingStock } from 'common/api/osrdEditoastApi';
 import { osrdMiddlewareApi, PowerRestrictionRange } from 'common/api/osrdMiddlewareApi';
 import { lengthFromLineCoordinates } from 'utils/geometry';
+import { setFailure } from 'reducers/main';
 
 type selectorOption = { key: string | undefined; value: string | undefined };
 
+type ElectrificationPR = {
+  [key: string]: string[] | (string | null | undefined)[];
```

Comment:
```suggestion
  [key: string]: string[] | (string)[];
```
Like this, it would be better :)

Reply:
Done